internal/driver/glfw/window.go

```diff
@@ -830,24 +830,18 @@ var keyCodeMap = map[glfw.Key]fyne.KeyName{
 	glfw.KeyMenu:         desktop.KeyMenu,
 }
 
-func keyToName(code glfw.Key) fyne.KeyName {
-	ret, ok := keyCodeMap[code]
-	if !ok {
-		return ""
-	}
-
-	return ret
-}
 
 func (w *window) keyPressed(viewport *glfw.Window, key glfw.Key, scancode int, action glfw.Action, mods glfw.ModifierKey) {
-	keyName := keyToName(key)
-	if keyName == "" {
+	keyCode, found := keyCodeMap[key]
+	if !found {
 		return
 	}
-	keyEvent := &fyne.KeyEvent{Name: keyName}
+	keyName := glfw.GetKeyName(key, scancode)
+
+	keyEvent := &fyne.KeyEvent{Name: keyCode}
```

Comment:
Thanks, this is clearer up to this point.
The KeyEvent "Name" should be the name, or we have not fixed the bug for other app developers.
It also does not fix the issue for custom keyboard shortcuts.

Reply:
Okay, this goes a bit beyond the scope of what I was trying to solve :P

I would suggest some refactoring:

change `type KeyName string` to `type KeyPosName string`

on `type CustomShortcut struct`:
- change `fyne.KeyName` to `fyne.KeyPosName`
- add `KeyName string`

on `type KeyEvent struct`:
- change `Name KeyName` to `Position KeyPosName`
- add `Name string`

This way, the code written based on the current situation doesn't break (it does change however). And for future use, developers can decide what is needed in their use case.

So, basically the diff below plus all necessary refactoring in the uses of `CustomShortcut` and `KeyEvent`

How should I continue? Both the issue and the pull request titles don't really represent the scope of the change we're talking about.

(By the way, I noticed `CustomShortcut` only supports 1 modifier key, while I use plenty of shortcuts that involve two modifier keys. But that should probably be put in another issue and pull request :P )

```diff
diff --git a/driver/desktop/shortcut.go b/driver/desktop/shortcut.go
index 0cc9da04..bd27fef1 100644
--- a/driver/desktop/shortcut.go
+++ b/driver/desktop/shortcut.go
@@ -12,7 +12,8 @@ var _ fyne.Shortcut = (*CustomShortcut)(nil)
 
 // CustomShortcut describes a shortcut desktop event.
 type CustomShortcut struct {
-	fyne.KeyName
+	fyne.KeyPosName // The name of the position of the key on the US_en keyboard
+	KeyName string  // The name of the key, i.e. the rune it would produce when pressed without modifiers
 	Modifier
 }
 
diff --git a/event.go b/event.go
index ddd62703..13ed617a 100644
--- a/event.go
+++ b/event.go
@@ -2,7 +2,8 @@ package fyne
 
 // KeyEvent describes a keyboard input event.
 type KeyEvent struct {
-	Name KeyName
+	Position KeyPosName // The name of the position of the key on the US_en keyboard
+	Name	 string     // The name of the key, i.e. the rune it would produce when pressed without modifiers
 }
 
 // PointEvent describes a pointer input event. The position is relative to the
diff --git a/internal/driver/glfw/window.go b/internal/driver/glfw/window.go
index 457e6dbc..f6c06f30 100644
--- a/internal/driver/glfw/window.go
+++ b/internal/driver/glfw/window.go
@@ -734,7 +734,7 @@ func convertMouseButton(btn glfw.MouseButton, mods glfw.ModifierKey) (desktop.Mo
 	return button, modifier
 }
 
-var keyCodeMap = map[glfw.Key]fyne.KeyName{
+var keyPosMap = map[glfw.Key]fyne.KeyPosName{
 	// non-printable
 	glfw.KeyEscape:    fyne.KeyEscape,
 	glfw.KeyEnter:     fyne.KeyReturn,
@@ -832,16 +832,19 @@ var keyCodeMap = map[glfw.Key]fyne.KeyName{
 
 
 func (w *window) keyPressed(viewport *glfw.Window, key glfw.Key, scancode int, action glfw.Action, mods glfw.ModifierKey) {
-	keyCode, found := keyCodeMap[key]
+	keyPosName, found := keyPosMap[key]
 	if !found {
 		return
 	}
 	keyName := glfw.GetKeyName(key, scancode)
 
-	keyEvent := &fyne.KeyEvent{Name: keyCode}
+	keyEvent := &fyne.KeyEvent{
+		Position: keyPosName,
+		Name: keyName,
+	}
 	keyDesktopModifier := desktopModifier(mods)
 
-	if keyCode == fyne.KeyTab {
+	if keyPosName == fyne.KeyTab {
 		if keyDesktopModifier == 0 {
 			if action != glfw.Release {
 				w.canvas.focusMgr.FocusNext(w.canvas.focused)
@@ -902,8 +905,9 @@ func (w *window) keyPressed(viewport *glfw.Window, key glfw.Key, scancode int, a
 	}
 	if shortcut == nil && keyDesktopModifier != 0 && keyDesktopModifier != desktop.ShiftModifier {
 		shortcut = &desktop.CustomShortcut{
-			KeyName:  keyCode,
-			Modifier: keyDesktopModifier,
+			KeyPosName: keyPosName,
+			KeyName:    keyName,
+			Modifier:   keyDesktopModifier,
 		}
 	}
 
diff --git a/key.go b/key.go
index a156145e..8a9da580 100644
--- a/key.go
+++ b/key.go
@@ -1,7 +1,7 @@
 package fyne
 
-// KeyName represents the name of a key that has been pressed
-type KeyName string
+// KeyName represents the name of the position of a key on a US_en keyboard
+type KeyPosName string
 
 const (
 	// KeyEscape is the "esc" key
```

Reply:
Thanks, I agree that this is a bigger issue than you realised - but I think we should fix it the right way rather than pushing a workaround into the library.

I don't want KeyName to become KeyPosName - these are supposed to describe semantic keys, not the specific position on a keyboard. The "KeyA" could refer to a virtual keyboard which has no concept of positions or hardware actions, so it needs to be a high level concept to work.

I think that the correct fix is that we need to lookup the Fyne key based on resolved key name from GLFW rather than the key code. Close to your original fix, but with a wider scope :)

Reply:
> (By the way, I noticed `CustomShortcut` only supports 1 modifier key, while I use plenty of shortcuts that involve two modifier keys. But that should probably be put in another issue and pull request :P )

Good catch. Please open an issue for this and I can fix for 1.2.4 - it's just a mistake in how the consts are defined, the API is OK otherwise I think. (i.e. `ControlModifier | ShiftModifier` should work, but it doesn't).

Reply:
I take that back - the modifiers do not have a bug, I misread the code - use 
`ControlModifier | ShiftModifier`.